homeassistant/setup.py

```diff
@@ -35,23 +35,23 @@
 
 BASE_PLATFORMS = {platform.value for platform in Platform}
 
-# DATA_SETUP is a dict[str, asyncio.Task[bool]], indicating domains which are currently
+# DATA_SETUP is a dict[str, asyncio.Future[bool]], indicating domains which are currently
 # being setup or which failed to setup:
 # - Tasks are added to DATA_SETUP by `async_setup_component`, the key is the domain
 #   being setup and the Task is the `_async_setup_component` helper.
 # - Tasks are removed from DATA_SETUP if setup was successful, that is,
 #   the task returned True.
 DATA_SETUP = "setup_tasks"
 
-# DATA_SETUP_DONE is a dict [str, asyncio.Future], indicating components which
+# DATA_SETUP_DONE is a dict [str, asyncio.Future[bool]], indicating components which
 # will be setup:
 # - Events are added to DATA_SETUP_DONE during bootstrap by
 #   async_set_domains_to_be_loaded, the key is the domain which will be loaded.
 # - Events are set and removed from DATA_SETUP_DONE when async_setup_component
 #   is finished, regardless of if the setup was successful or not.
 DATA_SETUP_DONE = "setup_done"
```

Comment:
I'd really like to collapse these into one `setup_tasks` but its hard to do with the existing design because they are filled inside a task